### PR TITLE
Enable Rubocop for Vagrantfile and docs/

### PR DESCRIPTION
**Pull request by eduardoj**
Enable Rubocop for Vagrantfile.

Enable Rubocop for `docs/dev/*` files and fix Rubocop complains.

`docs/api/restility/*` files are left in the `Exclude` section. These files must be fixed by Rubocop upstream, in [restility](https://github.com/openSUSE/restility) project.

Pair programmed with @dkang.

### CODE REVIEW

**Comment by eduardoj** [comment]
[After having Rubocop being enabled in base directory, enable rubocop for docs directory, and for Vagrantfile](https://trello.com/c/FuiWHVxV/323-after-having-rubocop-being-enabled-in-base-directory-enable-rubocop-for-docs-directory-and-for-vagrantfile)

File: docs/dev/model_template_example.rb [hunk 1]
```diff
@@ -18,8 +18,8 @@ class Dog < ActiveRecord::Base
   attr_accessor :number_of_barks
   attribute_method_prefix 'reset_'
 
-  alias_method :go_home, :save
-  alias_method :go_home!, :save!
+  alias go_home save
```
Comment (mdeniz) [inline docs/dev/model_template_example.rb]: Don't use `alias`, let it with `alias_method`. 
http://blog.bigbinary.com/2012/01/08/alias-vs-alias-method.html
http://stackoverflow.com/questions/4763121/should-i-use-alias-or-alias-method

File: docs/dev/model_template_example.rb [hunk 2]
```diff
@@ -18,8 +18,8 @@ class Dog < ActiveRecord::Base
...
+  alias go_home! save!
```
Comment (mdeniz) [inline docs/dev/model_template_example.rb]: Here to, let it with `alias_method`

File: .rubocop.yml [hunk 3]
```diff
@@ -8,7 +8,7 @@ AllCops:
     - 'src/api/tmp/**/*'
     - 'src/api/lib/templates/**/*'
     - 'src/api/vendor/bundle/**/*'
-    - 'docs/**/*'
+    - 'docs/api/restility/**/*'
```
Comment (Ana06) [inline .rubocop.yml]: why was this excluded?

Comment (eduardoj) [inline .rubocop.yml]: Please, have a look at the commit comment, or the pull request comment. It's already there. I will repeat it, for you:

`docs/api/restility/*` files are left in the `Exclude` section. These files must be fixed by Rubocop upstream, in [restility](https://github.com/openSUSE/restility) project.

File: docs/dev/model_template_example.rb [hunk 4]
```diff
@@ -75,15 +70,15 @@ def self.say(string)
...
+  #### Instance methods (public and then protected/private)
+  def initialize(attributes = {})
     super
     @number_of_barks = 0
   end
 
   def bark
     say('Guau!')
-    @number_of_barks++
+    @number_of_barks += 1
```
Comment (Ana06) [inline docs/dev/model_template_example.rb]: What cop did complain about this?

File: .rubocop.yml (hunk 3 above)
Comment (Ana06) [inline .rubocop.yml]: sorry, didn't read it :+1:

**Comment by Ana06** [comment]
@mdeniz 

> Don't use `alias`, let it with `alias_method`. 

This is funny because according to [Ruby style guide](https://github.com/bbatsov/ruby-style-guide#alias-method) `alias_method` should be used, but then the enforced style for `Style/Alias` is to used `alias` :joy: 

I just open an issue in Rubocop. :wink:

File: docs/dev/model_template_example.rb (hunk 4 above)
Comment (Ana06) [inline docs/dev/model_template_example.rb]: `++` doesn't exist in Ruby :rofl:

File: src/api/test/test_helper.rb [hunk 5]
```diff
@@ -170,7 +170,7 @@ def add_auth(headers)
         headers
       end
 
-      alias real_process process_with_kwargs
+      alias_method :real_process, :process_with_kwargs
```
Comment (bgeuken) [inline src/api/test/test_helper.rb]: Not long ago rubocop complained about using alias_method forced us to use alias. Funny that it's now the opposite.

**Comment by bgeuken** [comment]
I wonder if we shouldn't remove the example files... Anyway LGTM

**Pull request merged**